adminservice/src/intTest/groovy/nu/yona/server/ActivityCategoriesTest.groovy

```diff
@@ -27,5 +27,158 @@ class ActivityCategoriesTest extends Specification
+		String dutchName = "Schaken"
+		boolean isNoGo = true
+		def smoothwallCategories = ["chess"] as Set
+		def apps = ["Chess Free", "Analyze This", "Chess Opening Blunders"] as Set
+		String chessActivityCategoryJson = createActivityCategoryJson(["nl-NL": dutchName, "en-US" : englishName], isNoGo, smoothwallCategories, apps)
+		when:
+		def response = adminService.yonaServer.updateResource(createResponse.responseData._links.self.href, chessActivityCategoryJson)
+
+		then:
+		response.status == 200
+		response.responseData._links.self.href == createResponse.responseData._links.self.href
+		response.responseData.localizableName["en-US"] == englishName
+		response.responseData.localizableName["nl-NL"] == dutchName
+		response.responseData.mandatoryNoGo == isNoGo
+		response.responseData.smoothwallCategories as Set == smoothwallCategories
+		response.responseData.applications as Set == apps
+
+		def getResponse = adminService.yonaServer.getResource(createResponse.responseData._links.self.href)
+		getResponse.status == 200
+		getResponse.responseData._links.self.href == createResponse.responseData._links.self.href
+		getResponse.responseData.localizableName["en-US"] == englishName
+		getResponse.responseData.localizableName["nl-NL"] == dutchName
+		getResponse.responseData.mandatoryNoGo == isNoGo
+		getResponse.responseData.smoothwallCategories as Set == smoothwallCategories
+		getResponse.responseData.applications as Set == apps
+
+		cleanup:
+		if (createResponse.status == 200)
+		{
+			adminService.yonaServer.deleteResource(createResponse.responseData._links.self.href)
+		}
+	}
+
+	def 'Delete programming activity category' ()
+	{
+		given:
+		String programmingActivityCategoryJson = createActivityCategoryJson(["nl-NL": "Programmeren", "en-US" : "Programming"], false, ["programming", "scripting"], ["Eclipse", "Visual Studio"])
+		def createResponse = adminService.yonaServer.createResource(AdminService.ACTIVITY_CATEGORIES_PATH, programmingActivityCategoryJson)
+		assert createResponse.status == 200
+		def numActivityCategories = adminService.getAllActivityCategories().responseData._embedded."yona:activityCategories".size()
+		when:
+		def response = adminService.yonaServer.deleteResource(createResponse.responseData._links.self.href)
+
+		then:
+		response.status == 200
+		adminService.getAllActivityCategories().responseData._embedded."yona:activityCategories".size() == numActivityCategories - 1
+	}
+
+	def 'Try add duplicate English name' ()
+	{
+		given:
+		String programmingActivityCategoryJson = createActivityCategoryJson(["nl-NL": "Zomaar wat", "en-US" : "Gambling"], false, ["programming", "scripting"], ["Eclipse", "Visual Studio"])
+		when:
+		def response = adminService.yonaServer.createResource(AdminService.ACTIVITY_CATEGORIES_PATH, programmingActivityCategoryJson)
+
+		then:
+		response.status == 400
+		response.responseData.code == "error.activitycategory.duplicate.name"
+	}
+
+	def 'Try update to duplicate Dutch name' ()
```

Comment:
This is a different test. The previous one tries to **add** a duplicate name while the second one tries to **update** an existing causing a duplicate name. The first one uses English and the second Dutch.